### PR TITLE
Swagger module fixes for #2338, #2341 and #2342

### PR DESCRIPTION
**Pull request by omusil24**

### CODE REVIEW

**Comment by omusil24** [comment]
@davidnich do not merge yet, gonna fix #2341 too in this pull request.

**Comment by omusil24** [comment]
@davidnich you can merge now.

**Comment by omusil24** [comment]
@davidnich there's still a problem. swagger2service puts date as the type for the string/date values in the created hashdecls.

File: qlib/Swagger.qm [hunk 1]
```diff
@@ -345,7 +350,7 @@ class SchemaBase {
                         case "binary": return parse_hex_string(rv);
```
Comment (davidnich) [inline qlib/Swagger.qm]: actually I think for `case "string":` then here in `getExampleValue()` maybe nothing needs to be done - if we should return a string in any case, then we should just return the given value without processing.  Note that `rv` is already a string so calling `string(rv)` is a noop but with runtime overhead - meaning that the entire `case "string:"` block should be removed here - the binary parsing functions are also equally as wrong as the call to `date()` for example

File: qlib/Swagger.qm [hunk 2]
```diff
@@ -355,9 +360,9 @@ class SchemaBase {
             case "string": {
                 switch (format) {
                     case "byte":
-                    case "binary": return "<deadbeef>";
-                    case "date": return now_us();
-                    case "date-time": return now_us();
+                    case "binary": return <feedface>;
```
Comment (davidnich) [inline qlib/Swagger.qm:362]: this is the same as above - if we should return a string value, then why change it to a binary (`"<deadbeef>"` -> `<feedface>`) - if the date changes make sense, then I assume we need to continue to return a string also for the `binary` format

**Comment by omusil24** [comment]
@davidnich you can merge now.

**Review by davidnich — CHANGES_REQUESTED** [review]

File: qlib/Swagger.qm (hunk 2 above)
Comment (omusil24) [reply qlib/Swagger.qm:362]: IMO, for date it makes sense to leave it as a string to not mess it up with timezone mess but binary is a binary. It's not a string.

Comment (davidnich) [reply qlib/Swagger.qm:362]: I don't understand - if the method should return an example value, then returning a binary is correct for `binary` and returning a date is correct for `date` and `date-time`.  A date is also not a string.  If you mean that it should be a string anyway, then we can format it - like `now_us().format("YYYY-MM-DDTHH:mm:SS")` - but then I assume that the binary example should be returned as a string as well.

**Review by davidnich — APPROVED** [review]

**Pull request merged**